ui/brave_custom_notification/notification_control_buttons_view.cc

```diff
@@ -0,0 +1,118 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/ui/brave_custom_notification/notification_control_buttons_view.h"
+
+#include <memory>
+
+#include "brave/app/vector_icons/vector_icons.h"
+#include "brave/grit/brave_theme_resources.h"
+#include "brave/ui/brave_custom_notification/notification_view.h"
+#include "brave/ui/brave_custom_notification/padded_button.h"
+#include "brave/ui/brave_custom_notification/public/cpp/constants.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/resource/resource_bundle.h"
+#include "ui/compositor/layer.h"
+#include "ui/events/event.h"
+#include "ui/gfx/color_palette.h"
+#include "ui/gfx/paint_vector_icon.h"
+#include "ui/strings/grit/ui_strings.h"
+#include "ui/views/background.h"
+#include "ui/views/controls/image_view.h"
+#include "ui/views/controls/button/image_button.h"
+#include "ui/views/layout/box_layout.h"
+
+namespace brave_custom_notification {
+
+const char NotificationControlButtonsView::kViewClassName[] =
+    "NotificationControlButtonsView";
+
+NotificationControlButtonsView::NotificationControlButtonsView(
+    NotificationView* message_view)
+    : message_view_(message_view) {
+  DCHECK(message_view);
+  SetLayoutManager(std::make_unique<views::BoxLayout>(
+      views::BoxLayout::Orientation::kHorizontal));
+
+  // Use layer to change the opacity.
+  SetPaintToLayer();
+  layer()->SetFillsBoundsOpaquely(false);
+
+  SetBackground(views::CreateSolidBackground(kControlButtonBackgroundColor));
+}
+
+NotificationControlButtonsView::~NotificationControlButtonsView() = default;
+
+void NotificationControlButtonsView::ShowInfoButton(bool show) {
+  if (show && !info_button_) {
+    // Add the button next right to the snooze button.
+    // (Albert Wang) We change to an PaddedButton once we have a link
```

Comment:
If this is not required before release, can you please raise an issue and add a `TODO` referencing the issue. If not required, can you please remove